compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -576,12 +579,20 @@ static LogicalResult setTransposeConfig(func::FuncOp entryPoint,
   TileSizesListType tileSizes;
   tileSizes.push_back({tileM, tileN});
 
-  // Check alignment with tile size
+  // Check alignment with tile size for each transpose.
   if (auto genericOp = dyn_cast<linalg::GenericOp>(op)) {
-    auto inputShape =
-        genericOp.inputs()[0].getType().cast<ShapedType>().getShape();
-    if (inputShape[0] % tileM != 0 || inputShape[1] % tileN != 0) {
-      return failure();
+    for (auto operandIndexPair :
+         llvm::zip(genericOp.getOperands(), genericOp.getIndexingMapsArray())) {
+      if (isSharedMemTranspose(std::get<1>(operandIndexPair))) {
+        auto inputShape = std::get<0>(operandIndexPair)
+                              .getType()
+                              .cast<ShapedType>()
+                              .getShape();
+        if (inputShape[inputShape.size() - 1] % tileM != 0 ||
+            inputShape[inputShape.size() - 2] % tileN != 0) {
+          return failure();
```

Comment:
All the operands dimensions must be matching, instead you should just do `generic.getStaticLoopRanges()` and check that the dimension are divisible. Since you know there are only two dimensions you know it will get you M, N